### PR TITLE
Set Action to EXECUTION_FINISHED when job is done

### PR DESCRIPTION
**Pull request by vder**
## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala [hunk 1]
```diff
@@ -167,6 +173,16 @@ class PeriodicProcessService(
   }
 
   def handleFinished: Future[Unit] = {
+    def updateProcessAction(runningProcesses: Set[ProcessName])(
+        processName: ProcessName,
+        processActionIdOpt: Option[ProcessActionId]
+    ): Future[Boolean] = {
+      processActionIdOpt
+        .filter(_ => !runningProcesses(processName))
```
Comment (arkadius) [inline engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala:181]: Please rename this `runningProcesses` to sth like `stillActiveProcesses` or `processesWithRemainingRuns` or `stillScheduledProcesses`. Running here could be everyhing: running before `handleFisinhed` execution, some job deployed (RUNNING status) during computation of this method. Also, I'd rather like to keep this check near   `findActiveSchedulesForProcessesHavingDeploymentWithMatchingStatus` to be visible what's going on. Also, when we replace `DeploymentId` with `ProcessActionId`, the noise will be reduced.

File: engine/flink/management/periodic/src/test/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessServiceTest.scala [hunk 2]
```diff
@@ -68,6 +75,13 @@ class PeriodicProcessServiceTest
     val events                        = new ArrayBuffer[PeriodicProcessEvent]()
     val additionalData                = Map("testMap" -> "testValue")
 
+    implicit val deploymentService: ProcessingTypeDeploymentService with ActionIdChecker =
```
Comment (arkadius) [inline engine/flink/management/periodic/src/test/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessServiceTest.scala:78]: KISS. Why not just use concrete type (`ProcessingTypeDeploymentServiceStub`) instead?

File: designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ProcessingTypeDeploymentService.scala [hunk 3]
```diff
@@ -29,8 +29,11 @@ final case class DeployedScenarioData(
     resolvedScenario: CanonicalProcess
 )
 
-class ProcessingTypeDeploymentServiceStub(deployedScenarios: List[DeployedScenarioData])
-    extends ProcessingTypeDeploymentService {
+class ProcessingTypeDeploymentServiceStub(
+    deployedScenarios: List[DeployedScenarioData],
+) extends ProcessingTypeDeploymentService {
+
+  protected val actionIds: scala.collection.mutable.Buffer[ProcessActionId] = scala.collection.mutable.Buffer.empty
```
Comment (arkadius) [inline designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ProcessingTypeDeploymentService.scala:36]: It isn't thread safety and I'm not sure if we aren't doing something concurrently here. I'd recommend to replace this by `@volatile var` with immutable List

File: engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala [hunk 4]
```diff
@@ -188,9 +204,26 @@ class PeriodicProcessService(
...
+      nextSchedules <- scheduledProcessesRepository
+        .findActiveSchedulesForProcessesHavingDeploymentWithMatchingStatus(
+          Set(
+            PeriodicProcessDeploymentStatus.Deployed,
+            PeriodicProcessDeploymentStatus.Scheduled,
+            PeriodicProcessDeploymentStatus.RetryingDeploy
+          )
+        )
+        .run
+      runningProcesses = nextSchedules.groupByProcessName.keySet
```
[the remark this answers is not in the document]
Comment (vder) [reply engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala]: not applicable any more

File: engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala [hunk 5]
```diff
@@ -167,6 +173,16 @@ class PeriodicProcessService(
   }
 
   def handleFinished: Future[Unit] = {
+    def updateProcessAction(runningProcesses: Set[ProcessName])(
```
[the remark this answers is not in the document]
Comment (vder) [reply engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala]: renamed

File: engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala [hunk 6]
```diff
@@ -188,9 +204,26 @@ class PeriodicProcessService(
           Set(PeriodicProcessDeploymentStatus.Deployed, PeriodicProcessDeploymentStatus.FailedOnDeploy)
         )
         .run
-
+      schedulesToCheck = schedules.groupByProcessName.toList
       // we handle each job separately, if we fail at some point, we will continue on next handleFinished run
-      _ <- Future.sequence(schedules.groupByProcessName.toList.map(handleSingleProcess _ tupled))
+      _ <- Future.sequence(schedulesToCheck.map(handleSingleProcess _ tupled))
+      nextSchedules <- scheduledProcessesRepository
```
[the remark this answers is not in the document]
Comment (vder) [reply engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala]: ok  I've changed `reschedule` behaviour and removed some dead code in the process. Or at least it looked like a dead code to me :)

File: engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala (hunk 1 above)
Comment (vder) [reply engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala]: rescheduledProcesses?

**Review by arkadius — APPROVED** [review]

**Pull request merged**